addons/mail/models/mail_mail.py

```diff
@@ -221,8 +221,10 @@ def process_email_queue(self, ids=None):
         if 'filters' in self._context:
             filters.extend(self._context['filters'])
         # TODO: make limit configurable
-        filtered_ids = self.search(filters, limit=10000).ids
+        filtered_ids = self.search(filters, limit=batch_size if not ids else 10000).ids
         if not ids:
+            ids_count = len(filtered_ids) if len(filtered_ids) < batch_size else self.search_count(filters)
+            self.env['ir.cron']._notify_progress(0, ids_count)
```

Comment:
We plan to add an explicit ICP for mail queue, so would say this part is probably going to be rewritten anyway.

Reply:
This part is just to ensure that if it is re-triggered because of timeout it won't be considered as a failure.
I will go through every team CRON to apply new guidelines once this is merged